Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2_test.py

```diff
@@ -177,7 +139,197 @@ def test_build_readpdf_entry_object_empty_extract(mocker):
+
+            9. A PDF encoded with the second encoding, a text url ended without a slash /.
+
+            10. A PDF encoded with the second encoding, a text url without the http prefix.
+
+            11. A PDF encoded with the second encoding, a text email address.
+
+            12. A PDF encoded with the second encoding, a text url with an https prefix, and an email address in it.
+
+            13. A PDF encoded with the second encoding, a text hyperlink of a url.
+
+            14. A PDF encoded with the second encoding, aa embedded url (a url the is hyperlinked to an image).
+
+        When:
+            Running 'get_urls_and_emails_from_pdf_file' function on the PDF file.
+
+        Then:
+            Verify that the expected amount of URLs and Email addresses was extracted from the PDF.
+            Verify that the specific URL or Email was extracted successfully.
+
+    """
+    from ReadPDFFileV2 import get_urls_and_emails_from_pdf_file
+
+    # Select the PDF encoding:
+    if pdf_encoding_type == 1:  # first encoding type (a pdf created with google docs)
+        file_path = f'{CWD}/URLs_Extraction_Test_PDF_Encoding_Google_Docs_Renderer.pdf'
+    else:  # second encoding type (a pdf created with mac os notes)
+        file_path = f'{CWD}/URLs_Extraction_Test_PDF_Encoding_Quartz_PDFContext.pdf'
+
+    # Extract URLs and Emails:
+    urls = get_urls_and_emails_from_pdf_file(file_path)
+
+    assert len(urls) == 7
+    assert expected_output in urls
+
+
+def test_separate_urls_and_emails():
+    """
+        Given:
+        A set including urls and emails that were extracted from a PDF file.
+
+        When:
+            Running 'separate_urls_and_emails' function on the given set.
+
+        Then:
+            Verify the expected amount of URLs identified.
+            Verify the expected amount of Emails identified.
+            Verify that each url was classified as a url, ans email address was classified as an email.
+            Verify that the 'special url' that included both url and email was classified as a url, and the inner email
+            was classified as an email.
+    """
+    from ReadPDFFileV2 import separate_urls_and_emails
+
+    urls_and_emails_input_set = {'https://test.com/', 'www.test.net',
+                                 'https://test.com.co/ed/trn/update?email=user@test.net',
+                                 'mailto:user@testtest.com', 'user@testing.com'}
+
+    # Define the expected outputs:
+    expected_urls = ['https://test.com/', 'www.test.net', 'https://test.com.co/ed/trn/update?email=user@test.net']
+    expected_emails = ['user@testing.com', 'user@testtest.com', 'user@test.net']
+
+    # Separate URLs from Emails:
+    urls_ec, emails_ec = separate_urls_and_emails(urls_and_emails_input_set)
+    urls_ec = [item.get('Data') for item in urls_ec]
+    urls_ec_set = set(urls_ec)
+    emails_ec_set = set(emails_ec)
+
+    assert len(urls_ec) == 3
```

Comment:
I think we should do equals the set.
Meaning
```
url_set = (...)
email_set = (...)
eset, uset = separate_urls_and_emails( url_set | email_set)
assert eset == email_set
assert user == url_set
```

Reply:
Changed that.